mizar/dp/mizar/workflows/builtins/services/create.py

```diff
@@ -44,24 +44,37 @@ def requires(self):
 
     def run(self):
         logger.info("Run {task}".format(task=self.__class__.__name__))
-        net = nets_opr.store.get_net(OBJ_DEFAULTS.default_ep_net)
+        net = net_opr.store.get_net(OBJ_DEFAULTS.default_ep_net)
+        if self.param.body['metadata'].get('annotations'):
+            if self.param.body['metadata'].get('annotations').get(OBJ_DEFAULTS.mizar_ep_vpc_annotation):
+                vpc_name = self.param.body['metadata'].get(
+                    'annotations').get(OBJ_DEFAULTS.mizar_ep_vpc_annotation)
+                vpc = vpc_opr.store_get(vpc_name)
+                if not vpc:
+                    self.raise_temporary_error(
+                        "VPC {} for service {} does not exist!".format(vpc_name, self.param.name))
+                if self.param.body['metadata'].get('annotations').get(OBJ_DEFAULTS.mizar_ep_subnet_annotation):
+                    subnet_name = self.param.body['metadata'].get(
+                        'annotations').get(OBJ_DEFAULTS.mizar_ep_subnet_annotation)
+                    subnet = net_opr.store.get_net(subnet_name)
+                    if subnet.vpc != vpc_name:
+                        self.raise_temporary_error("Subnet {} of pod {} does not belong to VPC {}".format(
+                            subnet_name, self.param.name, vpc_name))
+                    if not subnet:
+                        self.raise_temporary_error(
+                            "Subnet {} of pod {} does not exist!".format(subnet_name, self.param.name))
+                else:
+                    subnets = list(net_opr.store.get_nets_in_vpc(vpc_name))
+                    if subnets:
+                        subnet_name = subnets[0]
+                        logger.info("Subnet not specified, allocating pod {} in subnet {} for VPC {}".format(
```

Comment:
Do you intend for this log message to say "Subnet specified" instead of "Subnet not specified"?

Reply:
Ah, thanks for the catch!